test/scripts/setup-osbuild-repo

```diff
@@ -3,27 +3,50 @@
 # Add a repository configuration to install the osbuild rpm at a specific
 # commit if specified.
 import os
+import sys
+import urllib.error
+import urllib.request
 
 import imgtestlib as testlib
 
 REPO_TEMPLATE = """
 [osbuild]
 name=osbuild {commit}
-baseurl=http://osbuild-composer-repos.s3-website.us-east-2.amazonaws.com/{repo_path}
+baseurl={baseurl}
 enabled=1
 gpgcheck=0
 priority=10
 """
 
 REPO_FILE = "/etc/yum.repos.d/osbuild.repo"
+URL_TEMPLATE = "http://osbuild-composer-repos.s3-website.us-east-2.amazonaws.com/{repo_path}"
+
+
+def baseurl(repo_path):
+    return URL_TEMPLATE.format(repo_path=repo_path)
+
+
+def check_baseurl(repo_baseurl):
+    repomd_url = repo_baseurl + "/repodata/repomd.xml"
+    try:
+        with urllib.request.urlopen(repomd_url, timeout=3) as resp:
+            print(f"{resp.status} ({resp.msg})")
+    except urllib.error.HTTPError as http_error:
+        print(http_error)
+        sys.exit(1)
 
 
 def write_repo(commit, distro_version):
     arch = os.uname().machine
     repo_path = f"osbuild/{distro_version}/{arch}/{commit}"
+    repo_baseurl = baseurl(repo_path)
+
+    print(f"Checking dnf repository for {commit} ({repo_path})")
```

Comment:
Move the "/repodata/repomd.xml" bit down here and log the full path being tested (or move the logging into `check_baseurl` -- if I see an error the first thing I'm going to do is copy and paste the url into my browser or curl and test it so it should match what's actually failing.

Reply:
You make an excellent point.  Fixed!